File: src/Samples/CSharpApp/ViewModels/OrderViewModel.cs
namespace CSharpApp.ViewModels
{
    using System.ComponentModel;
    using Arcadia;
    using Arcadia.MVVM;
    using Data;
    using Models;

    public class OrderViewModel : PageViewModel
    {
        public OrderViewModel(IOrderCalculationEngine calculationEngine) : base()
        {
            Model = calculationEngine;
            AddOrderItemCommand = new ActionCommand(OnAddOrderItem);
        }

        public override string Name { get { return "Order"; } }

        public IOrderCalculationEngine Model { get; private set; }
        public ActionCommand AddOrderItemCommand {get; private set;}

        private void OnAddOrderItem()
        {
            Model.Order.Items.AddNew();
        }
    }
}

File: src/Samples/FSharpApp/ViewModels/OrderViewModel.fs
namespace FSharpApp.ViewModels

open System.ComponentModel
open System.Linq
open Arcadia
open Arcadia.MVVM
open FSharpApp.Models

type OrderViewModel(ce : IOrderCalculationEngine) = 
    inherit PageViewModel()
    let model = ce

    member this.Model : IOrderCalculationEngine = ce

    override this.Name = "Order"

    member this.AddOrderItemCommand = ActionCommand(fun() -> this.Model.Order.Items.AddNew() |> ignore)
